cluster/storage-cluster.go

```diff
@@ -304,3 +304,71 @@ func createTenantInStorageGroup(ctx *Context, tenant *Tenant, sg *StorageGroup)
 	}()
 	return ch
 }
+
+// Returns a list of tenants that are allocated to the provided `StorageGroup`
+func GetTenantStorageGroupByShortName(ctx *Context, tenantShortName string) chan *StorageGroupTenantResult {
+	ch := make(chan *StorageGroupTenantResult)
+	go func() {
+		defer close(ch)
+		if tenantShortName == "" {
+			ch <- &StorageGroupTenantResult{Error: errors.New("empty tenant short name")}
```

Comment:
we should return immediately after sending the error on the channel.
```suggestion
			ch <- &StorageGroupTenantResult{Error: errors.New("empty tenant short name")}
			return
```